frontend/src/metabase/components/ItemTypeFilterBar.jsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { Flex } from "grid-styled";
+import { t } from "c-3po";
+import { withRouter } from "react-router";
+
+import Link from "metabase/components/Link";
+
+import colors from "metabase/lib/colors";
+
+export const FILTERS = [
+  {
+    name: t`Everything`,
+    filter: null,
+  },
+  {
+    name: t`Dashboards`,
+    filter: "dashboard",
+  },
+  {
+    name: t`Questions`,
+    filter: "card",
+  },
+  {
+    name: t`Pulses`,
+    filter: "pulse",
+  },
+];
+
+const ItemTypeFilterBar = props => {
+  const { location } = props;
+  return (
+    <Flex align="center" className="border-bottom">
+      {props.filters.map(f => {
+        let isActive = location.query.type === f.filter;
+
+        if (!location.query.type && !f.filter) {
+          isActive = true;
+        }
+
+        const color = isActive ? colors.brand : "inherit";
+
+        return (
+          <Link
+            to={{
+              pathname: location.pathname,
+              query: Object.assign({}, location.query, {
```

Comment:
FYI `Object.assign` is basically the same as `{ ...location.query, type: f.filter }`